core/src/test/java/com/linecorp/armeria/internal/common/PathAndQueryTest.java

```diff
@@ -162,6 +165,84 @@ void doubleDotsInNameValueQuery() {
         });
     }
 
+    private static String normalizedQuery(String string) {
```

Comment:
Ideally, I think we want to use a variant of `PathAndQuery#decodePercentsAndEncodeToUtf8` in case this implementation goes stale. I think it's fine to leave if it's too much work though